Rubberduck.Parsing/SelectionExtensions.cs

```diff
@@ -0,0 +1,55 @@
+﻿using Antlr4.Runtime;
+using Rubberduck.VBEditor;
+using System;
+
+namespace Rubberduck.Parsing
+{
+    /// <summary>
+    /// Provide extensions on selections & contexts/tokens
+    /// to assist in validating whether a selection contains
+    /// a given context or a token. 
+    /// </summary>
+    public static class SelectionExtensions
+    {
+        /// <summary>
+        /// Validates whether a token is contained within a given Selection
+        /// </summary>
+        /// <param name="selection">One-based selection, usually from CodePane.Selection</param>
+        /// <param name="token">An individual token within a module's parse tree</param>
+        /// <returns>Boolean with true indicating that token is within the selection</returns>
+        public static bool Contains(this Selection selection, IToken token)
+        {
+            return
+                (((selection.StartLine == token.Line) && (selection.StartColumn - 1) <= token.Column) 
+                    || (selection.StartLine < token.Line))
+             && (((selection.EndLine == token.EndLine()) && (selection.EndColumn - 1) >= (token.EndColumn())) 
+                    || (selection.EndLine > token.EndLine()));
+        }
+
+        /// <summary>
+        /// Validates whether a context is contained within a given Selection
+        /// </summary>
+        /// <param name="context">A context which contains several tokens within a module's parse tree</param>
+        /// <param name="selection">One-based selection, usually from CodePane.Selection</param>
```

Comment:
nitpick: xml-doc params should be in the same order as the actual signature :wink: